staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/grpc_service.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package kmsv2 transforms values for storage at rest using a Envelope provider
+package kmsv2
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"google.golang.org/grpc"
+
+	"k8s.io/apiserver/pkg/storage/value/encrypt/envelope/util"
+	kmsapi "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v2alpha1"
+)
+
+// The gRPC implementation for envelope.Service.
+type gRPCService struct {
+	kmsClient   kmsapi.KeyManagementServiceClient
+	connection  *grpc.ClientConn
+	callTimeout time.Duration
+}
```

Comment:
Is there ever going to be two implementations of this? Just call this Service and export it directly (i.e. delete Service interface).

Reply:
I don't _think_ we'll have another implementation of this. The way it was structured for v1 with the `Service` interface enables us to test the `value.Transformer` interface thoroughly. IMO, we could lose some of that if we just rename this and remove the `Service` interface. LMK if this is valid or if we should still go ahead and just make the change.